### DISCUSSION TITLE
Websocket TLS handshake error EOF

### DISCUSSION THREAD

**Discussion by game5413** [nats-io/nats-server#2919] [Q&A]
Hello, currently i'm trying to setup nats websocket with tls based example configuration on this [reference](https://docs.nats.io/running-a-nats-service/configuration/websocket/websocket_conf) on my local environment, tls that i'm use was self signing certificate from mkcert, then mounting them into docker container alongside with my nats server configuration.

nats-server.conf
```
# Client port of 4222 on all interfaces
port: 4222

# HTTP monitoring port
monitor_port: 8222

authorization {
	default_permissions: {
		publish: "SANDBOX.*"
		subscribe: ["PUBLIC.>", "_INBOX.>"]
	}
	ADMIN: {
		publish: ">"
		subscribe: ">"
	}
	WS_USER: {
		publish: ["cmd.req", "_INBOX.>"]
		subscribe: ["notifications.*", "_INBOX.>"]
	}
	users: [
		{ user: admin, password: "admin", permissions: $ADMIN, allowed_connection_types: ["STANDARD"] }
		{ user: user, pass: "user", permissions: $WS_USER, allowed_connection_types: ["WEBSOCKET"] }
	]
	timeout: 15.0
}

# This is for clustering multiple servers together.
cluster {
  name: "executive-nats-cluster"

  port: 6222

  authorization {
    user: "cUser"
    password: "cPassword"
    timeout: 2
  }

  routes = []
}

websocket {
    port: 6969

    tls {
          cert_file: "/etc/nats/mkcert-server-cert.pem"
          key_file: "/etc/nats/mkcert-server-key.pem"
    }

    authorization {
		timeout: 3.0
    }
}
```
With config above, i'm tried connecting to websocket using [nats.ws](https://github.com/nats-io/nats.ws) as client, with example code like below, and got an error message unknown error

![image](https://user-images.githubusercontent.com/37659721/157583054-1be11514-3072-4e2e-aa51-36c8cd6b012e.png)

after that, i'm checking container logs on docker and see an error logs that say `TLS handshake error EOF` like on picture below

![image](https://user-images.githubusercontent.com/37659721/157582074-2e89fb0c-2d49-48e3-9fe3-6e6d8370d44f.png)

Is there something wrong with SSL Certificate or Nats server configuration ?

Environment:

- Windows 10 Pro 21H2
- WSL2
- Docker Desktop

**Comment by kozlovic** [comment]
I suggest that we close this discussion since I see that you have reported an issue in nats.ws repo, and mention that you can connect using the nats.go client, so looks more like a library issue?

**Reply by game5413** [reply]
Yeah, it seem issue from the library, thanks for your response.
